### PR TITLE
Feat/452 step 3 validation fe

### LINKED ISSUE #452: Step 3 Validation FE
Criteria:

- [ ] Validation of user inputs
- [ ] Ensure all mandatory fields are filled in

### PR DESCRIPTION
**Pull request by mgaseta**
# Description
Closes #452 #698

#### New
- Added validation rules for step 3
- Synced collectors agency and location code to interim agency and location code
- Added new rules to the progress bar to display the correct status for interim step

#### Changed
- The UI of the page:
Before: 
Screenshot from 2023-12-05 19-27-38
Now:
Screenshot from 2023-12-05 19-28-51

#### Removed
- Unnecessary fields (see Changed section)

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-697-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-697-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-697-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-697-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-697-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-697-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/components/ApplicantAgencyFields/index.tsx [hunk 1]
```diff
@@ -63,12 +63,14 @@ const ApplicantAgencyFields = ({
 
     const updatedAgency = {
       ...agency,
-      value: checked ? defaultAgency : EmptyMultiOptObj
+      value: checked ? defaultAgency : EmptyMultiOptObj,
+      isInvalid: !!(checked && defaultAgency?.label === '')
```
Comment (craigyu) [inline frontend/src/components/ApplicantAgencyFields/index.tsx:67]: out of curiosity, why do we need to cast here

Comment (mgaseta) [reply frontend/src/components/ApplicantAgencyFields/index.tsx:67]: We don't, I did a ESlint fix, it came with the cast and I forgot to remove... Will fix that.

Comment (mgaseta) [reply frontend/src/components/ApplicantAgencyFields/index.tsx]: Fixed here: https://github.com/bcgov/nr-spar/pull/697/commits/aab22da3958eb4c5aa1c61e16216f5e122aac64a

Comment (craigyu) [reply frontend/src/components/ApplicantAgencyFields/index.tsx]: works!

**Pull request merged**